File: 2015 maj/c++/3.cpp
#include <iostream>
#include <fstream>
#include <algorithm>
using namespace std;


int main()
{
    ifstream in("liczby.txt");

    string s[1001];
    int smallest = 1, biggest = 1;
    for (int line = 1; line <= 1000; line++)
    {
        in >> s[line];

        // check if smaller
        if (s[line].size() < s[smallest].size())
            smallest = line;
        else if (s[line].size() == s[smallest].size())
        {
            for (int i = 0; i < s[line].size(); i++)
            {
                if (s[line][i] != s[smallest][i])
                {
                    if (s[line][i] == '0')
                        smallest = line;
                    break;
                }
            }
        }

        // check if bigger
        if (s[line].size() > s[biggest].size())
            biggest = line;
        else if (s[line].size() == s[biggest].size())
        {
            for (int i = 0; i < s[line].size(); i++)
            {
                if (s[line][i] != s[biggest][i])
                {
                    if (s[line][i] == '1')
                        biggest = line;
                    break;
                }
            }
        }
    }
    cout << smallest << ' ' << biggest << '\n';
}

File: 2015 maj/python/3.py
with open("liczby.txt") as f:
    s = f.read().strip().split('\n')

smallest, biggest = 0, 0
for i in range(1, 1000):

    """
        Jeżeli liczby są różnej długości to mniejsza jest ta krótsza, a większa jest ta dłuższa
        Jeżeli liczby są tej samej długości to mniejsza ma 0 na pierwszym różniącym się dla porównywanych słów bicie od lewej
    """

    if len(s[i]) < len(s[smallest]):
        smallest = i
    elif len(s[i]) == len(s[smallest]):
        for j in range(len(s[i])):
            if s[i][j] != s[smallest][j] and s[i][j] == '0':
                smallest = i
                break

    if len(s[i]) > len(s[biggest]):
        biggest = i
    elif len(s[i]) == len(s[biggest]):
        for j in range(len(s[i])):
            if s[i][j] != s[biggest][j] and s[i][j] == '1':
                biggest = i
                break

print(smallest + 1, biggest + 1)